R/data-set.R

```diff
@@ -0,0 +1,277 @@
+#' Supported types of the error
+#' @include enum.R
+#' @export
+DataErrorType <- enum(c(
+  "ArithmeticStdDev",
+  "GeometricStdDev"
+))
+
+#' @title DataSet
+#' @docType class
+#' @description  A class for storage of numerical x- and y-value pairs and optional error for y-values.
+#' @export
+#' @format NULL
+DataSet <- R6::R6Class(
+  "DataSet",
+  inherit = DotNetWrapper,
+  cloneable = FALSE,
+  active = list(
+    #' @field name The name of the DataSet
+    name = function(value) {
+      if (missing(value)) {
+        return(private$.dataRepository$name)
+      }
+      private$.dataRepository$name <- value
+    },
+    #' @field xDimension Dimension in which the xValues are defined
+    xDimension = function(value) {
+      if (missing(value)) {
+        return(private$.xColumn$dimension)
+      }
+      private$.setColumnDimension(private$.xColumn, value)
+    },
+    #' @field xUnit Unit in which the xValues are defined
+    xUnit = function(value) {
+      if (missing(value)) {
+        return(private$.xColumn$displayUnit)
+      }
+      private$.setColumnUnit(private$.xColumn, value)
+    },
+    #' @field xValues Values stored in the xUnit. This field is read-only.
+    xValues = function(values) {
+      if (missing(values)) {
+        return(private$.getColumnValues(private$.xColumn))
+      }
+
+      private$throwPropertyIsReadonly("xValues")
+    },
+    #' @field yDimension Dimension in which the yValues are defined
+    yDimension = function(value) {
+      if (missing(value)) {
+        return(private$.yColumn$dimension)
+      }
+      private$.setColumnDimension(private$.yColumn, value)
+    },
+    #' @field yUnit Unit in which the yValues are defined
+    yUnit = function(value) {
+      if (missing(value)) {
+        return(private$.yColumn$displayUnit)
+      }
+      private$.yColumn$displayUnit <- value
+    },
+    #' @field yValues Values stored in the yUnit. This field is read-only.
+    yValues = function(values) {
+      if (missing(values)) {
+        return(private$.getColumnValues(private$.yColumn))
+      }
+      private$throwPropertyIsReadonly("yValues")
+    },
+
+    #' @field yErrorType Type of the error - geometric or arithmetic.
+    #' When changing from arithmetic to geometric error, the values are considered in as fraction (1 = 100%).
+    #' When changing from geometric to arithmetic, the values are set to the same unit as \code{yErrorUnit}.
+    yErrorType = function(value) {
+      private$.createErrorColumnIfMissing()

```

Comment:
when accessing Error properties, I create the column if it's not there. The way it was implemented @PavelBal , you would modify the original repo if loaded from file and I don't think this is pure

Reply:
I don't really understand... Now the original repo is modified the first time any property of yError is called, no?